core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/base/BaseAuthorizationRule.scala

```diff
@@ -98,9 +97,9 @@ private[auth] trait BaseAuthorizationRule
 
   private def authorizeLoggedUser[B <: BlockContext : BlockContextUpdater](blockContext: B,
                                                                            user: LoggedUser,
-                                                                           userGroupsProvider: (B, LoggedUser, Set[GroupIdLike]) => Task[UniqueList[Group]]): Task[RuleResult[B]] = {
-    if (blockContext.isCurrentGroupEligible(groupsPermittedByRule)) {
-      userGroupsProvider(blockContext, user, groupsPermittedByRule.groupIds.toSet)
+                                                                           userGroupsProvider: (B, LoggedUser) => Task[UniqueList[Group]]): Task[RuleResult[B]] = {
+    if (blockContext.isCurrentGroupPotentiallyEligible(groupsLogic)) {
```

Comment:
I wonder if in this place we don't have the same issue we had in the case of the `BaseGroupsRule` class. We checked if the current group is potentially eligible but not if it's really eligible in the case of the "not" group logic.

Could you please write a test to prove the issue (or prove there is no such issue) too?

Reply:
Yes, there was an issue with `preferredGroupId`, again not detected by tests. I added 8 tests in `LdapAuthorizationRuleTests`, two of them specifically for this case.